api/app/GraphQL/Mutations/SwapAssessmentStepOrder.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\AssessmentStep;
+use Nuwave\Lighthouse\Exceptions\ValidationException;
+
+final class SwapAssessmentStepOrder
+{
+    /**
+     * @param  null  $_
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        $stepA = AssessmentStep::find($args['stepA']);
+        $stepB = AssessmentStep::find($args['stepB']);
+
+        // Ensure the steps belong to the same pool
+        if ($stepA->pool_id !== $stepB->pool_id) {
+            throw ValidationException::withMessages(['AssessmentSteps must belong to the same pool.']);
```

Comment:
Any i18n strings should be moved to `packages/i18n/src/messages/apiMessages.ts` so that they can be translated.